doc/source/whatsnew/v0.18.0.txt

```diff
@@ -507,6 +507,32 @@ Subtraction by ``Timedelta`` in a ``Series`` by a ``Timestamp`` works (:issue:`1
    pd.Timestamp('2012-01-01') - ser
 
 
+Changes to msgpack
+^^^^^^^^^^^^^^^^^^
+
+Forward incompatible changes in ``msgpack`` writing format were made over 0.17.0 and 0.18.0; older versions of pandas cannot read files packed by newer versions (:issue:`12129`, `10527`)
+
+Bug in ``to_msgpack`` and ``read_msgpack`` introduced in 0.17.0 and fixed in 0.18.0, caused files packed in Python 2 unreadable by Python 3 (:issue:`12142`)
+
+.. warning::
+
+As a result of a number of issues:
+
+      +----------------------+------------------------+
+      | Packed with          | Can be unpacked with   |
+      +======================+========================+
+      | pre-0.17 / Python 2  | any                    |
+      +----------------------+------------------------+
+      | pre-0.17 / Python 3  | any                    |
+      +----------------------+------------------------+
+      | 0.17 / Python 2      | - 0.17 / Python 2      |
+      |                      | - >=0.18 / any Python  |
+      +----------------------+------------------------+
+      | 0.17 / Python 3      | >=0.18 / any Python    |
```

Comment:
so maybe say (and in io.rst as well). that 0.18.0 is backward compat for reading (except for 0.17/py2)